### PR TITLE
openstack: add simple test for the openstack-populator

### PR DESCRIPTION
**Pull request by bennyz**
* This PR also changes the populator controller and ovirt-populator to use `envFrom` instead of a file to pass the secret.

This will eventually go into the CDI controller PR, but is also useful here for now

### CODE REVIEW

File: cmd/ovirt-populator/ovirt-populator.go [hunk 1]
```diff
@@ -156,50 +156,30 @@ func populate(engineURL, diskID, volPath string) {
...
+	var insecureSkipVerify string
+	_, found := os.LookupEnv("insecureSkipVerify")
+	if !found {
+		insecureSkipVerify = "false"
 	} else {
-		insecureSkipVerify, err = os.ReadFile("/etc/secret-volume/insecureSkipVerify")
-		if err != nil {
-			klog.Fatal(err.Error())
-		}
+		insecureSkipVerify = os.Getenv("insecureSkipVerify")
```
Comment (ahadas) [inline cmd/ovirt-populator/ovirt-populator.go:167]: why not taking the value from the call to `LookupEnv` above as well?

**Review by ahadas — COMMENTED** [review]
nice! minor comment inside
ah and SonarCloud has a point in the code smell it reports..

**Comment by ahadas** [comment]
> ah and SonarCloud has a point in the code smell it reports..

@bennyz so what about that tiny part?

**Review by liranr23 — COMMENTED** [review]
what's the exact benefit? will you see secret in the env of the pod?

File: pkg/lib-volume-populator/populator-machinery/controller.go [hunk 2]
```diff
@@ -628,11 +628,7 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 					},
 				}
 			}
-			con.VolumeMounts = append(con.VolumeMounts, corev1.VolumeMount{
```
[the remark this answers is not in the document]
Comment (bennyz) [reply pkg/lib-volume-populator/populator-machinery/controller.go:631]: what do you mean?

[the remark this answers is not in the document]
Comment (liranr23) [reply pkg/lib-volume-populator/populator-machinery/controller.go:631]: we use `SecretName: secretName,`, meaning we have a secret available in the namespace. do we use it elsewhere? if not - should we remove the creation of it?

**Review by ahadas — APPROVED** [review]

**Comment by ahadas** [comment]
> what's the exact benefit? will you see secret in the env of the pod?

yeah, it's a cleaner [way to get data from secret](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#define-container-environment-variables-using-secret-data)

**Pull request merged**